internal/ingress/controller/nginx.go

```diff
@@ -658,21 +658,21 @@ Error: %v
 // changes were detected. The received backend Configuration is merged with the
 // configuration ConfigMap before generating the final configuration file.
 // Returns nil in case the backend was successfully reloaded.
-func (n *NGINXController) OnUpdate(ingressCfg ingress.Configuration) error {
+func (n *NGINXController) OnUpdate(ingressCfg *ingress.Configuration) error {
```

Comment:
I'm not sure I feel comfortable with changing this to a pointer vs a full copy of the structure as it was. While it is a good optimization, OnUpdate shouldn't have write access to the cfg, and the cfg shouldn't be mutated by other processes while being rendered.